### DISCUSSION TITLE
Zero values

### DISCUSSION THREAD

**Discussion by DrownFish19** [PaddlePaddle/PaddleSpatial#29] [BaiduKDDCup2022] [answered]
In the BaiduKDDCup2022_report, the zero values need to be processed as follows:
> Zero values. There are some active power and reactive power which are smaller than zeros. We simply treat all the values which are smaller than 0 as 0, i.e. if 𝑃𝑎𝑡𝑣 < 0, then 𝑃𝑎𝑡𝑣 = 0.

However, the zero values are not processed in the baseline.

Moreover, the missing values are set as zero, then the missing values and unknown values are not calculated in the RMSE and MAE, as follows:
```python
# for missing values
df_data.replace(to_replace=np.nan, value=0, inplace=True) 
preds = predictions[np.where(grounds != 0)]
gts = grounds[np.where(grounds != 0)]
```

```python
# for unknown values
cond = (raw_data['Patv'] <= 0) & (raw_data['Wspd'] > 2.5) | \
           (raw_data['Pab1'] > 89) | (raw_data['Pab2'] > 89) | (raw_data['Pab3'] > 89)
indices = np.where(~cond[i:out_seq_len + i])
prediction = prediction[indices]
targets = targets[indices]
```

 Q: Are the zero values not calculated in the RMSE and MAE?

**Comment by ramber1836** [comment, marked as answer]
The current baseline code does not handle zero values, which will be updated accordingly in the next version dedicated for final evaluation.

**Reply by yuhao-morning** [reply]
So it means that we need to discard all the zero values, missing values and unknown values of Patv when doing evaluation?

**Comment by ramber1836** [comment]
Apologies for the late reply. 
Please refer to the newly released baseline scripts for the details.

**Comment by yuhao-morning** [comment]
report上说对missing value, unknown value and abnormal value 都认为预测值和真实值是一样的（也就是对应数据的metrics为0），但是代码上是把这些值对应的数据都去掉了再算metrics, 哪一个是准确的？










At 2022-05-05 21:13:42, "Xinjiang Lu" ***@***.***> wrote:

Apologies for the late reply.
Please refer to the newly released baseline scripts for the details.

**Comment by yuhao-morning** [comment]
report上对小于0的Patv 的处理是全部变为0， 但是过滤条件是raw_data['Patv'] <= 0) & (raw_data['Wspd'] > 2.5) ， 所以对于小于0的 Patv, 如果不满足这个过滤条件，那么它的值是0吗？而且这个数据是需要被evaluation的吗
















At 2022-05-05 21:13:42, "Xinjiang Lu" ***@***.***> wrote:

Apologies for the late reply.
Please refer to the newly released baseline scripts for the details.

**Reply by ramber1836** [reply]
Please refer to the newly released baseline code

**Comment by yuhao-morning** [comment]
所以，代码里是将na先转为0， 所以这跟report很不一样，report是说Patv为NA（Missing）则将它不计为算 metrics的数据，这个东西怎么会一直不清楚？？所以我们也是要按代码里的先把NA变为0，然后根据cond过滤掉不算做metrics的数据吗？代码里没有将Patv负数变为0啊，这跟report又不一样，这到底是怎么回事么？我们要保持Patv为负，还是先将它变为0再处理呢？













At 2022-05-07 15:48:17, "Xinjiang Lu" ***@***.***> wrote:

Please refer to the newly released baseline code
